src/Build/Definition/ProjectProperty.cs

```diff
@@ -110,6 +115,27 @@ string IProperty.EvaluatedValueEscaped
             }
         }
 
+        /// <summary>
+        /// Gets object's location in xml file.
+        /// </summary>
+        public (string File, int Line, int Column) Location { get => _location; }
+
+        string IProperty.GetEvaluatedValueEscaped(IElementLocation location)
+        {
+            if (this is EnvironmentDerivedProjectProperty environmentProperty && environmentProperty.loggingContext is { IsValid: true } loggingContext && !environmentProperty._loggedEnvProperty && !Traits.LogAllEnvironmentVariables)
```

Comment:
From reading the condition: 
`!Traits.LogAllEnvironmentVariables` => If the logAllEnvVars  trait is not enabled then log the env variable? 
However didn't check the actual value of the trait, just clarifying if I have correct understanding

Reply:
We have some rules defined around the logging environment variables described here:
https://github.com/dotnet/msbuild/blob/fb681758a40376e93cd0c316274d4aeeda599f69/documentation/wiki/Binary-Log.md?plain=1#L43

Briefly saying: this condition helps to report the events that influence MSBuild specifically.

Reply:
Got it thank you! misunderstanding raised from the naming of the Trait itself, not it is clear thanks :)